### PR TITLE
Add image support

### PR DESCRIPTION
**Pull request by rajanmaghera**
Note: This PR is dependent on #52 

This PR adds support to create and display image posts. Other than that, images behave like normal posts.

There is a new model called ImageFile, which is used as an internal representation of the image. When we need it to be displayed in base64, we only then encode it as so for the API response. To accomplish this, I define a new property on Post called content_formatted. This will automatically turn an ImageFile into base64 and vice-versa when setting.

There is also support for the API route /image after a post url. To make this work, I added a new route in DRF.

To use the images from the front-end templates, I defined a new template inclusion. At the top of your file, include the line `{% load image_extras %}`. Then, anywhere in the file, include `{% image post %}` where `post` is the name of a post, to insert an image tag with the correct source URL.

Screenshot 2023-03-02 at 2 21 08 PM

### CODE REVIEW

File: quickcomm/api.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+
+# This file houses the API views.
+
+from django.http import FileResponse
+from django.shortcuts import get_object_or_404
+from rest_framework import viewsets
+from rest_framework.response import Response
+from rest_framework.decorators import action
```
Comment (Jdvdb) [inline quickcomm/api.py:8]: I don't think you use this import

File: quickcomm/forms.py [hunk 2]
```diff
@@ -57,7 +64,49 @@ def save(self, author):
         post.save()
         return post
 
+class CreateImageForm(forms.Form):
+    title = forms.CharField(max_length=100)
```
Comment (Jdvdb) [inline quickcomm/forms.py:68]: maybe define this like you have for other classes

File: quickcomm/forms.py [hunk 3]
```diff
@@ -57,7 +64,49 @@ def save(self, author):
...
+        # Save the image to the media folder
+        image = ImageFile(
+            post=post,
+            image=self.cleaned_data['content']
+        )
+        image.save()
+
+        return post
 class CreateLoginForm(forms.Form):
+    """A form for logging in."""
```
Comment (Jdvdb) [inline quickcomm/forms.py:109]: thank you for adding this for me <3

File: quickcomm/serializers.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+from rest_framework import serializers
+from rest_framework_nested.relations import NestedHyperlinkedRelatedField
```
Comment (Jdvdb) [inline quickcomm/serializers.py:2]: I don't think this is being used anymore

File: quickcomm/templatetags/image_extras.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+from django import template
+from django.template import Context
```
Comment (Jdvdb) [inline quickcomm/templatetags/image_extras.py:2]: it doesn't look like this uses Context

File: quickcomm/tests/test_api_authors.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
+
+from django.test import TestCase
+from rest_framework.test import APIRequestFactory, force_authenticate, APIClient
```
Comment (Jdvdb) [inline quickcomm/tests/test_api_authors.py:3]: force_authenticate is not needed

File: quickcomm/tests/test_api_authors.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        self.assertEqual(req.status_code, 405)
+
+    def test_get_author(self):
+        """Test that we can get an author"""
+        req = self.client.get('/api/authors/{}/'.format(self.author.id))
+        self.assertEqual(req.status_code, 200)
+        self.assertEqual(req.data['type'], 'author')
+        self.assertNotEquals(req.data['type'], 'authors')
+
+        self.assertEqual(req.data['id'], str(self.author.id))
```
Comment (Jdvdb) [inline quickcomm/tests/test_api_authors.py:56]: id issue still present in this test (i imagine there's a merge conflict for this)

File: quickcomm/forms.py (hunk 3 above)
Comment (rajanmaghera) [reply quickcomm/forms.py:109]: ofc

File: quickcomm/tests/test_api_authors.py (hunk 7 above)
Comment (rajanmaghera) [reply quickcomm/tests/test_api_authors.py:56]: Yes, I changed this, but another PR must have overwritten this.

**Pull request merged**